### PR TITLE
House robber -- solution(Dynamic programming approach)

### PR DESCRIPTION
**Pull request by ShahriarCodes**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have added a LeetCode problem [House Robber](https://leetcode.com/problems/house-robber/) solution in **Python**. I used dynamic programming approach to solve the problem. The directory to the solution is [NeoAlgo/Python/cp/](https://github.com/TesseractCoding/NeoAlgo/tree/master/Python/cp)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
The issue number is #1312 and can be found [here](https://github.com/TesseractCoding/NeoAlgo/issues/1312)

### CODE REVIEW

**Comment by ShahriarCodes** [comment]
Please review this pull request. @SKAUL05

File: Python/cp/house_robber.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            return 0
+        elif len(nums) == 1: 
+            return nums[0]
+        
+
+        runningTotal = [-1,-1]
+        runningTotal[0] = nums[0]
+        runningTotal[1] = max(nums[0], nums[1])
+
+        for i in range(2, len(nums)):
```
Comment (SKAUL05) [inline Python/cp/house_robber.py]: Change i to index

**Review by SKAUL05 — CHANGES_REQUESTED** [review]

File: Python/cp/house_robber.py (hunk 1 above)
Comment (ShahriarCodes) [reply Python/cp/house_robber.py]: @SKAUL05 
changed i to index. Please check the [new commit](https://github.com/TesseractCoding/NeoAlgo/pull/1315/commits/7040544f067acbe84d152e736360f4a829ed7edf)

**Review by SKAUL05 — APPROVED** [review]

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shahriar100 Format your code according to PEP 8 rules. You can refer to this http://pep8online.com/.

File: Python/cp/house_robber.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        runningTotal[1] = max(nums[0], nums[1])
+
+        for index in range(2, len(nums)):
+            runningTotal.append(max([nums[index] + runningTotal[index - 2], runningTotal[index - 1]]))
+
+
+        return runningTotal[-1]
+
+if __name__ == '__main__':
+    nums = [1,2,3,1]
```
Comment (atarax665) [inline Python/cp/house_robber.py]: Dynamic input required.

**Comment by ShahriarCodes** [comment]
@iamrajiv  Please check the [commit](https://github.com/TesseractCoding/NeoAlgo/pull/1315/commits/32902ef0cebcf5181468eb21de434944f5caa795) now. It passed all the test on **PEP8 online checker**

File: Python/cp/house_robber.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class Solution(object):
```
Comment (atarax665) [inline Python/cp/house_robber.py:1]: No need of making a class for this, function would be fine.

File: Python/cp/house_robber.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        for index in range(2, len(nums)):
+            runningTotal.append(max([nums[index] + runningTotal[index - 2], runningTotal[index - 1]]))
+
+
+        return runningTotal[-1]
+
+if __name__ == '__main__':
+    nums = [1,2,3,1]
+    instance = Solution()
+    print(instance.rob(nums))
```
Comment (atarax665) [inline Python/cp/house_robber.py]: Add complexity of your program and sample input/output in commented form.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/cp/house_robber.py (hunk 4 above)
Comment (ShahriarCodes) [reply Python/cp/house_robber.py]: @atarax665 Please check the [commit](https://github.com/TesseractCoding/NeoAlgo/pull/1315/commits/6c003d24be03e79e3026859b3a62a6091125483f) now.

**Review by iamrajiv — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]
LGTM :)

**Pull request merged**